R/util.R

```diff
@@ -610,3 +610,46 @@ abbreviate <- function(x, len = round(getOption("width", 80) * 0.8)) {
   }
   x
 }
+
+handle_missing_packages <- function(missing_packages) {
+  ## collapse vector to packages to string "c('pckg_1','pckg_2')"
+  vector_packages <- sprintf("install.packages(c(%s))",
+                             paste(sprintf("'%s'", missing_packages),
+                                   collapse=","))
+
+  ## check if we are interactive and logging is active...
+  logging <- !isTRUE(getOption("orderly.nolog"))
+  if (interactive() && logging) {
+    ## ...if so ask if Orderly should try to install the pacakges
+    question <- "Should I try to install missing packages by running:"
+    install_command <- paste0("\n", question, "\n\n",
+                              "    ", vector_packages)
+
+    try_install <- (utils::menu(c("no", "yes"), FALSE,
+                               title = install_command) == 2)
+
+    if (try_install) {
+      for (pckg in missing_packages) {
+        orderly_log("INSTALL",
+                    sprintf("Attempting to install %s", pckg))
+        withCallingHandlers(install.packages(pckg, quiet = TRUE),
```

Comment:
less errorprone (because `install.packages` is almost impossible to work around) given we know that these are not currently installed, is to just check that after running the install they are found.